### ISSUE TITLE
Support for conservative depth

### ISSUE THREAD

**Issue by Sixmorphugus** [gfx-rs/wgpu#3961]
**Is your feature request related to a problem? Please describe.**
Yes. I am using wgpu to make a native application that would greatly benefit from use of the [optional conservative depth feature of OpenGL](https://www.khronos.org/opengl/wiki/Fragment_Shader#Conservative_Depth), which [is also available in Vulkan via the SPIR-V execution mode](https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#Execution_Mode) but this feature does not yet seem to be available in wgpu or in wgsl.

This feature is for fragment shaders. It affects the [early depth test](https://www.khronos.org/opengl/wiki/Early_Fragment_Test), causing it to be able to discard more fragments by allowing it to make the assumption that the depth output from a fragment shader that writes to depth will always be greater or less than the fixed function depth your shader will write if you do not manually write anything to the output fragment depth value (gl_FragDepth in glsl, or the frag_depth builtin output in wgsl). For example, you could discard fragments for faces that write to depth even if they are obscured if you know they will always write greater depth and never less.

**Describe the solution you'd like**
I am making my own Teardown-style raymarcher and it would be very helpful to have a way to turn DepthGreater on while still using wgsl for the shaders.

**Describe alternatives you've considered**
I have considered using GLSL instead but it seems like giving up on an (imo) much better shader language because of 1 missing feature. 

**Additional context**
None

**Comment by Wumpf** [comment]
Related to these Naga issues:
* https://github.com/gfx-rs/wgpu/issues/4408
* https://github.com/gfx-rs/wgpu/issues/4410

Afaik there's some partial support already: https://github.com/gfx-rs/naga/pull/2255
but it's not exposed in wgpu.

**Comment by Sixmorphugus** [comment]
Does that mean there's no (hacky) way to turn it on?

**Comment by Wumpf** [comment]
No, but it's practically just waiting for someone to tie together the loose ends and expose it in wgpu as a native-only feature :)
